presto-main/src/test/java/io/prestosql/server/TestServer.java

```diff
@@ -192,6 +194,32 @@ public void testNoTransactionSupport()
         assertEquals(queryResults.getError().getErrorCode(), INCOMPATIBLE_CLIENT.toErrorCode().getCode());
     }
 
+    @Test(dataProvider = "testVersionOnErrorDataProvider")
+    public void testVersionOnError(String query)
+    {
+        QueryResults queryResults = postQuery(request -> request
+                .setBodyGenerator(createStaticBodyGenerator(query, UTF_8)))
+                .map(JsonResponse::getValue)
+                .filter(result -> result.getError() != null)
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Error expected"));
+
+        assertNull(queryResults.getNextUri());
+        QueryError queryError = queryResults.getError();
+        assertThat(queryError.getFailureInfo().toException())
+                .hasStackTraceContaining("at io.prestosql.$gen.Presto_null__testversion____");
+    }
+
+    @DataProvider
+    public Object[][] testVersionOnErrorDataProvider()
+    {
+        return new Object[][] {
+                {"SELECT query that fails parsing"}, // fails during parsing
+                {"SELECT foo FROM no.such.table"}, // fails during analysis
+                {"SELECT 1 / 0"}, // fails during optimization
```

Comment:
can you also add test that fails during execution: `select 1 / a from (values 0) t(a)`?

Reply:
thanks, added